### ISSUE TITLE
Speculative decode draft model produces 0% acceptance (placeholder weights)

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#938]
## Draft Model Training Status

The draft model training completed successfully:
- **10 epochs** (3370 steps) on 5400 samples
- **Checkpoint**: `/home/bcloud/checkpoints/1bit-spec/eagle3_qwen3_0.6b/step_3370/`
- **Exported binary**: `/home/bcloud/spec-decode/checkpoints/eagle3_draft_v3.bin` (1.3 GB)
- **Forward pass**: Valid non-NaN logits, correct vocabulary indices

## Acceptance Rate

**0%** — the 1-layer Eagle3 draft model does not predict the same argmax tokens as the 28-layer target. This is expected for a model this small. Typical speculative decoding requires a draft model with ~5-10% of the target's parameters to achieve 60-80% acceptance.

## Recommendations to Improve Acceptance

1. **Increase draft model size**: Change `draft_num_hidden_layers` from 1 to 2-3 in the config
2. **More training data**: Increase from 5700 to 50k+ samples
3. **More epochs**: Train for 30+ epochs instead of 10
4. **Use DFlash or DSpark architecture**: These use 5 layers instead of 1 and have higher acceptance

## Running Training with New Config

```bash
source ~/spec-decode/train-venv2/bin/activate
cd ~/spec-decode

# Edit config to increase draft layers:
sed -i 's/draft_num_hidden_layers=1/draft_num_hidden_layers=2/' configs/eagle3_qwen3_0.6b.py
sed -i 's/exp_name = "eagle3_qwen3_0.6b"/exp_name = "eagle3_qwen3_0.6b_v2"/' configs/eagle3_qwen3_0.6b.py

RANK=0 LOCAL_RANK=0 WORLD_SIZE=1 MASTER_ADDR=127.0.0.1 MASTER_PORT=29502 \
CUDA_VISIBLE_DEVICES=0 \
python3 ~/DeepSpec/train.py \
    --config configs/eagle3_qwen3_0.6b.py \
    --opts train.lr=0.0006 \
    --opts train.num_train_epochs=30
```
